### ISSUE TITLE
Canonical orders returned by Traces inconsistent

### ISSUE THREAD

**Issue by juliuskunze** [a-maier/nauty-Traces-sys#8]
Hi @a-maier,

I wrote a test verifying that two small isomorphic node-labeled graphs give consistent canonical orders with Traces:

https://github.com/a-maier/nauty-Traces-sys/compare/master...juliuskunze:nauty-Traces-sys:traces-bug

Both graphs are isomorphic to the following structure consisting of two disconnected parts (colors highlight the node labels):

$${\color{red}0} - {\color{green}1} - {\color{green}1}$$

$${\color{red}0} - {\color{green}1} - {\color{green}1}$$

The test fails because the canonical orders are inconsistent and the canonicalized graphs are different.

Traces' canonical order is consistent for most of the ~200000 graphs I tested on but inconsistent for ~10 out of them.  sparsenauty is prohibitively slow in many of these cases, so this issue is critical for my use case.

**Comment by a-maier** [comment]
Hi @juliuskunze,

Thanks for reporting this bug. I can also reproduce it without the bindings:

https://github.com/a-maier/traces-canon-issue/blob/main/traces_canon_issue.c

It looks like a genuine Traces issue to me. It's probably best if you get in touch with the authors.

**Comment by juliuskunze** [comment]
Thanks for reproducing this directly in C! I contacted the authors and will let you know here if there are any updates.

**Comment by a-maier** [comment]
This should now be fixed in version 0.7.0, bundling nauty & Traces 2.8.8.

**Comment by juliuskunze** [comment]
Hey @a-maier, while the example now works, I just found that if you swap the order in which partitions are specified (essentially swapping the node labels 0 to 1 and 1 to 0), Traces still breaks:

In [my bindings test](https://github.com/a-maier/nauty-Traces-sys/compare/master...juliuskunze:nauty-Traces-sys:traces-bug#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759):
```rust
let mut lab = vec![5, 4, 2, 1, 3, 0]; // instead of vec![3, 0, 5, 4, 2, 1]
let mut ptn = vec![1, 1, 1, 0, 1, 0]; // instead of vec![1, 0, 1, 1, 1, 0]
...
let mut lab_ = vec![4, 3, 2, 0, 5, 1]; // instead of vec![5, 1, 4, 3, 2, 0]
```

Or in [your C test](https://github.com/a-maier/traces-canon-issue/blob/main/traces_canon_issue.c):
```C
static const int LAB[] = {5, 4, 2, 1, 3, 0}; // instead of {3, 0, 5, 4, 2, 1}
...
static const int PTN[] = {1, 1, 1, 0, 1, 0}; // instead of {1, 0, 1, 1, 1, 0}
...
static const int LAB[] = {4, 3, 2, 0, 5, 1}; // instead of {5, 1, 4, 3, 2, 0}
```

I didn't open a new this issue because it's not about the bindings. I will contact the authors - this is just for your information and amusement.

**Comment by juliuskunze** [comment]
Can confirm that this is resolved in nauty/Traces 2.9.3 used in version 0.11.
